bin/setup.csh

```diff
@@ -0,0 +1,46 @@
+# setup lsstsw environment
+#
+# source this file from your ~/.cshrc
+#
+# possibly blank, but can't be parsed as $_
+set cmd=($_)
+if ( $?0 ) then
+    set source=$0
+else if ( $#cmd >= 3 ) then
+    set source=`pwd`/${cmd[2]}
+else if (-f /usr/sbin/lsof ) then
+    set source=`/usr/sbin/lsof +p $$ | grep -oE /.\*setup.csh`
```

Comment:
You may find in the future that you really don't want to have the setup script sourced as part of your log in. I define a couple of shell aliases to let me initialize different `lsstsw` build trees (mainly for python versions) in different shells (switching trees later on is hard as you have to work out which environment variables to clean up).

Remember that `newinstall` creates the `loadLSST.*` setup scripts which are run to initialize the LSST environment. The fact that the code is in `newinstall` which is called once is not relevant to the discussion.

This is the logic from `loadLSST.csh` that is equivalent to what is being attempted here:
```csh
               set sourced=(\$_)
                if ("\${sourced}" != "") then
                   # If not already initialized, set LSST_HOME to the directory where this script is located
                   set this_script = \${sourced[2]}
                   if ( ! \${?LSST_HOME} ) then
                          set LSST_HOME = \`dirname \${this_script}\`
                          set LSST_HOME = \`cd \${LSST_HOME} && pwd\`
                   endif
```